v2/yarpcprotobuf/protoc-gen-yarpc-go/internal/generator/registry.go

```diff
@@ -0,0 +1,293 @@
+		target, ok := r.files[name]
+		if !ok {
+			return fmt.Errorf("file target %q was not registered", name)
+		}
+		for _, s := range target.descriptor.GetService() {
+			if err := r.loadService(target, s); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// GetData returns the template data the corresponds
+// to the given filename.
+func (r *registry) GetData(filename string) (*Data, error) {
+	f, err := r.getFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	return &Data{
+		File:    f,
+		Imports: r.imports,
+	}, nil
+}
+
+// getFile returns the File that corresponds to the
+// given filename.
+func (r *registry) getFile(filename string) (*File, error) {
+	f, ok := r.files[filename]
+	if !ok {
+		return nil, fmt.Errorf("file %q was not found", filename)
+	}
+	return f, nil
+}
+
+// getMessage returns the Message that corresponds to the
+// given name. This method expects the input to be formed
+// as an input or output type, such as .foo.Bar.
+func (r *registry) getMessage(name string) (*Message, error) {
+	// All input and output types are represented as
+	// .$(Package).$(Message), so we explicitly trim
+	// the leading '.' prefix.
+	msg := strings.TrimPrefix(name, ".")
+	m, ok := r.messages[msg]
+	if !ok {
+		return nil, fmt.Errorf("message %q was not found", msg)
+	}
+	return m, nil
+}
+
+// loadFile registers the given file's message types.
+// Note that we load the messages for all files up-front so that
+// all of the message types potentially referneced in the proto
+// services can reference these types.
+func (r *registry) loadFile(f *descriptor.FileDescriptorProto) {
+	pkg := r.newPackage(f)
+	file := &File{
+		descriptor: f,
+		Name:       f.GetName(),
+		Package:    pkg,
+	}
+	r.files[file.Name] = file
+	for _, m := range f.GetMessageType() {
+		r.loadMessage(file, m)
+	}
+}
+
+func (r *registry) loadMessage(f *File, m *descriptor.DescriptorProto) {
+	name := m.GetName()
+	msg := &Message{
+		Name:    name,
+		Package: f.Package,
+	}
+	r.messages[f.Package.fqn(name)] = msg
+
+	for _, n := range m.GetNestedType() {
+		r.loadMessage(f, n)
+	}
+}
+
+func (r *registry) loadService(f *File, s *descriptor.ServiceDescriptorProto) error {
+	name := s.GetName()
+	svc := &Service{
+		Name:       name,
+		FQN:        f.Package.fqn(name),
+		Client:     join(name, _yarpc, _client),
+		ClientImpl: join(_implPrefix, name, _yarpc, _client),
+		FxClient:   join(_fx, name, _yarpc, _client),
+		Server:     join(name, _yarpc, _server),
+		ServerImpl: join(_implPrefix, name, _yarpc, _server),
+		FxServer:   join(_fx, name, _yarpc, _server),
+		Procedures: join(name, _yarpc, _procedures),
+	}
+	for _, m := range s.GetMethod() {
+		method, err := r.newMethod(m, name)
+		if err != nil {
+			return err
+		}
+		svc.Methods = append(svc.Methods, method)
+	}
+	f.Services = append(f.Services, svc)
+	return nil
+}
+
+func (r *registry) newMethod(m *descriptor.MethodDescriptorProto, svc string) (*Method, error) {
```

Comment:
nit: let's use `service` instead of `svc` as a variable (assuming no naming conflicts). `svc` is not consistent with the rest of the naming in the repo.